xbmc/rendering/dx/DeviceResources.cpp

```diff
@@ -1484,3 +1484,20 @@ bool DX::DeviceResources::SetMultithreadProtected(bool enabled) const
 
   return (wasEnabled == TRUE ? true : false);
 }
+
+bool DX::DeviceResources::IsGCNOrOlder() const
+{
+  if (CSysInfo::GetWindowsDeviceFamily() == CSysInfo::Xbox)
+    return false;
+
+  if (GetAdapterDesc().VendorId != PCIV_AMD)
+    return false;
+
+  VideoDriverInfo driver = DX::DeviceResources::Get()->GetVideoDriverVersion();
+
+  if (driver.part[0] > 31 || (driver.part[0] == 31 && driver.part[1] > 0) ||
+      (driver.part[0] == 31 && driver.part[1] > 0 && driver.part[2] >= 22000))
```

Comment:
`driver.part[0] == 31 && driver.part[1] > 0` seems redundant with previous condition

You mean `(driver.part[0] == 31 && driver.part[1] == 0 && driver.part[2] >= 22000)` ?

Reply:
I don't think so, eventually there will be a new WDDM with 32.xx.xx.xx versions, I don't know if there will ever be 31.1, ...
I'd like to compact the expression but not sure how in a safe way.

That issue was side-stepped in IsWindowsVersionAtLeast() by assigning our own identifiers, which is not doable here.
Maybe a dedicated function to test IsVersionAtLeast() would make this more presentable.

Reply:
(A & B) | (A & B & C) is same as (A & B) but is not what I'm saying. For me seems a typo and current code:

```c++
if (driver.part[0] > 31 || (driver.part[0] == 31 && driver.part[1] > 0) ||
      (driver.part[0] == 31 && driver.part[1] > 0 && driver.part[2] >= 22000))
```

should be:

```c++
if (driver.part[0] > 31 || (driver.part[0] == 31 && driver.part[1] > 0) ||
      (driver.part[0] == 31 && driver.part[1] == 0 && driver.part[2] >= 22000))
```

Reply:
I see. Right, that was a typo. I replaced with a reusable function, which could be even more general, but that will be for later, if we need that driver version comparison somewhere else.